xbmc/GUIPassword.cpp

```diff
@@ -445,13 +444,13 @@ bool CGUIPassword::CheckMenuLock(int iWindowID)
       break;
     case WINDOW_MUSIC_NAV:      // Music
       bCheckPW = profileManager->GetCurrentProfile().musicLocked();
-      if (!bCheckPW && strMediasourcePath != "") // check mediasource by path
-        return g_passwordManager.IsMediaPathUnlocked(strMediasourcePath, "music");
+      if (!bCheckPW && !GetMediaSourcePath().empty()) // check mediasource by path
```

Comment:
I don't get why you have introduced the getter method, as `m_strMediaSourcePath` is never read from outside class `CGUIPassword` and you don't need to mutex protect access to this member. So, no additional encapsulation is needed. So, what is the benefit of this additional code?

Reply:
you're right, there is no benefit. the reason is i am not an experienced c++ programmer. was just looking at my first pr ever (which would never be there without help) and trying to tune it a little. call it practicing ;)
setting a public var (now `m_strMediaSourcePath`) from outside looked somehow odd to me, too. so i thought about getter and setter.
if you say this makes it all more complicated, i have no problem with dropping it.
> i'm not sure if all changes should be kept 

(just tell me)

Reply:
imo 072afd22d8142556110b4a790f65ee07d2407db6 would be a valid usecase for the setter.

Reply:
> i am not an experienced c++ programmer. 

 You are welcome and we are happy to assist you.

> setting a public var (now m_strMediaSourcePath) from outside looked somehow odd to me (in this context). 

 And you are right with this. public member vars break encapsulation (on of the basic principles of OOP).

> so i thought about getter and setter.

 The setter is fine, for the reason given above. The getter is not needed here as there is no access from outside the class.

Reply:
>>   i'm not sure if all changes should be kept

> (just tell me)

 Keep the setter, remove the getter.

Reply:
thanks - on the way